source/MaterialXGenShader/ShaderStage.h

```diff
@@ -104,6 +105,9 @@ class VariableBlock
     /// Add an existing shader port to this block.
     void add(ShaderPortPtr port);
 
+    /// Find a port based on a predicate
+    ShaderPort* findByPredicate(const ShaderPortPredicate& predicate);
```

Comment:
Could be overloaded and named `find` now that the signature is different from find by name.

Reply:
Nickpick: it could then be placed under the other find methods.